RELEASE.md

```diff
@@ -0,0 +1,51 @@
+# Releasing Kubeflow Pipelines
+
+WIP: this document is still incomplete.
+
+## Common Prerequisites
+
+* Permissions needed
+    * Can create a branch in github.com/kubeflow/pipelines
+* Tools that should be in your `$PATH`
+    * jq 1.6 https://stedolan.github.io/jq/download/
+    * yq https://github.com/mikefarah/yq/releases/tag/3.3.0
+    * jdk 8
+* Preparations
+    1. Clone github.com/kubeflow/pipelines repo into `$KFP_REPO`
+    2. `cd $KFP_REPO`
+
+## Cutting a release branch
+
+1. Choose a good commit on master branch with commit hash as `$COMMIT_SHA`
+2. Choose the next release branch's `$MINOR_VERSION` in format `x.y`, e.g. `1.0`, `1.1`...
+2. Make a release branch of format `release-$MINOR_VERSION`, e.g. `release-1.0`, `release-1.1`. Branch from the commit and push to kubeflow pipelines upstream repo.
+    ```bash
+    git checkout $COMMIT_SHA
+    git checkout -b release-$MINOR_VERSION
+    git push upstream HEAD
+    ```
+
+## Releasing from release branch
+
+1. Choose the release's complete `$VERSION` following semantic versioning, e.g.
+    * `1.0.0-rc.1`
+    * `1.0.0-rc.2`
+    * `1.0.0`
+    * `1.0.1`
+    * ...
+1. Update all version refs in release branch by
+    ```bash
+    ./hack/release.sh $VERSION release-$MINOR_VERSION
+    ```
+    It will prompt you whether to push it to release branch. Press `y` and hit `Enter`.
+
+    Note, the script will clone kubeflow/pipelines repo into a temporary location on your computer, make those changes and attempt to push to upstream, so that it won't interfere with your current git repo.
+
+    TODO: this script should also regenerate
+    * changelog
+    * python api client
+1. Wait and make sure the cloudbuild job that builds all images in gcr.io/ml-pipeline-test succeeded for above commit. Then submit the second cloudbuild job that copies these images to gcr.io/ml-pipeline.
```

Comment:
> Then submit the second cloudbuild job that copies these images to gcr.io/ml-pipeline
would you detail more on how to submit the second job?

I'm not sure whether you mean it's triggered by GCP-Github hook, or we manually trigger it via run script in hack folder?

Reply:
The easiest way right now is to go to gcp UI and click cloudbuild history and retry the failed task triggered by adding the new version tag. I'll refine this part when I reach a final solution.